terraform/ecs_ec2/daemon/main.tf

```diff
@@ -63,6 +63,12 @@ resource "aws_autoscaling_group" "cluster" {
     value               = ""
     propagate_at_launch = true
   }
+
+  tag {
+    key                 = "BaseClusterName"
+    value               = local.basename
+    propagate_at_launch = true
+  }
```

Comment:
nit - since the purpose of this new tag is to identify resources that can be "cleaned" I suggest the key name be "AutomaticCleanerGroup" or "AutomaticCleanerTag" (or anything that makes it clear to a random person that the tag is related to automatically cleaning up resources.

And then we could just look for the presence of the tag name. Setting the value might be useful if we want to just clean up resources for certain tests or test suites...